### PR TITLE
fix: rebuilding in volume.meta blocks engine start

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/6626

### CODE REVIEW

File: pkg/backend/remote/remote.go [hunk 1]
```diff
@@ -266,6 +275,40 @@ func (r *Remote) SetUnmapMarkSnapChainRemoved(enabled bool) error {
...
+	if !isRebuilding {
+		return nil
+	}
+
+	logrus.Warnf("Resetting %v rebuild", r.name)
+
+	conn, err := grpc.Dial(r.replicaServiceURL, grpc.WithInsecure(),
+		ptypes.WithIdentityValidationClientInterceptor(r.volumeName, ""))
+	if err != nil {
+		return fmt.Errorf("cannot connect to ReplicaService %v: %v", r.replicaServiceURL, err)
```
Comment (derekbit) [inline pkg/backend/remote/remote.go:293]: errors.Wrapf(err, "....")

File: pkg/controller/control.go [hunk 2]
```diff
@@ -737,6 +737,13 @@ func (c *Controller) Start(volumeSize, volumeCurrentSize int64, addresses ...str
 			continue
 		}
 
+		if len(addresses) == 1 {
+			err = newBackend.ResetRebuild()
```
Comment (derekbit) [inline pkg/controller/control.go:741]: Can we add a comment explaining the purpose of the reset of rebuilding flag for single replica volume?

**Review by derekbit — COMMENTED** [review]
In general,LGTM.

**Review by innobead — APPROVED** [review]
LGTM, since this is for one replica case. However, this is not for fixing the root cause, but rather an resilience solution to recover from the failure (which could be caused by some concourrent operations between replica rebuilding)

**Review by derekbit — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@mergify backport v1.5.x v1.4.x

**Pull request merged**